src/FormElementManager.php

```diff
@@ -29,54 +29,94 @@
 use function sprintf;
 
 /**
- * laminas-servicemanager v2-compatible plugin manager implementation for form elements.
+ * laminas-servicemanager v3-compatible plugin manager implementation for form elements.
  *
  * Enforces that elements retrieved are instances of ElementInterface.
  */
-class FormElementManagerV2Polyfill extends AbstractPluginManager
+class FormElementManager extends AbstractPluginManager
```

Comment:
Oh, didn't see that.
So you've renamed the v2 plugin manager instead of the v3 one.

Please diff with the v3 plugin manager from 2.15.0 to see whats missing in here.

Reply:
>  So you've renamed the v2 plugin manager instead of the v3 one.

No, I've renamed the v3 one, don't know why git tells otherwise. It already is BC, no API changes introduced except for `protected $shareByDefault = false;` which can't work anyway with v3:

```diff
$ git diff 2.16.x:src/FormElementManager/FormElementManagerV3Polyfill.php php_80_support:src/FormElementManager.php
diff --git a/src/FormElementManager/FormElementManagerV3Polyfill.php b/src/FormElementManager.php
index aaa8202d..e9af6a3a 100644
--- a/src/FormElementManager/FormElementManagerV3Polyfill.php
+++ b/src/FormElementManager.php
@@ -6,10 +6,11 @@
  * @license   https://github.com/laminas/laminas-form/blob/master/LICENSE.md New BSD License
  */

-namespace Laminas\Form\FormElementManager;
+namespace Laminas\Form;

 use Interop\Container\ContainerInterface;
 use Laminas\Form\Element;
+use Laminas\Form\Exception;
 use Laminas\Form\ElementFactory;
 use Laminas\Form\ElementInterface;
 use Laminas\Form\Fieldset;
@@ -32,10 +33,8 @@ use function sprintf;
  *
  * Enforces that elements retrieved are instances of ElementInterface.
  */
-class FormElementManagerV3Polyfill extends AbstractPluginManager
+class FormElementManager extends AbstractPluginManager
 {
-    use FormElementManagerTrait;
-
     /**
      * Aliases for default set of helpers
      *
@@ -277,13 +276,6 @@ class FormElementManagerV3Polyfill extends AbstractPluginManager
      */
     protected $sharedByDefault = false;

-    /**
-     * Don't share form elements by default (v2)
-     *
-     * @var bool
-     */
-    protected $shareByDefault = false;
-
     /**
      * Interface all plugins managed by this class must implement.
      * @var string
@@ -399,4 +391,62 @@ class FormElementManagerV3Polyfill extends AbstractPluginManager

         return $this;
     }
+    /**
+     * Try to pull hydrator from the creation context, or instantiates it from its name
+     *
+     * @param  string $hydratorName
+     * @return mixed
+     * @throws Exception\DomainException
+     */
+    public function getHydratorFromName($hydratorName)
+    {
+        $services = $this->creationContext;
+
+        if ($services && $services->has('HydratorManager')) {
+            $hydrators = $services->get('HydratorManager');
+            if ($hydrators->has($hydratorName)) {
+                return $hydrators->get($hydratorName);
+            }
+        }
+
+        if ($services && $services->has($hydratorName)) {
+            return $services->get($hydratorName);
+        }
+
+        if (! class_exists($hydratorName)) {
+            throw new Exception\DomainException(sprintf(
+                'Expects string hydrator name to be a valid class name; received "%s"',
+                $hydratorName
+            ));
+        }
+
+        $hydrator = new $hydratorName;
+        return $hydrator;
+    }
+
+    /**
+     * Try to pull factory from the creation context, or instantiates it from its name
+     *
+     * @param  string $factoryName
+     * @return mixed
+     * @throws Exception\DomainException
+     */
+    public function getFactoryFromName($factoryName)
+    {
+        $services = $this->creationContext;
+
+        if ($services && $services->has($factoryName)) {
+            return $services->get($factoryName);
+        }
+
+        if (! class_exists($factoryName)) {
+            throw new Exception\DomainException(sprintf(
+                'Expects string factory name to be a valid class name; received "%s"',
+                $factoryName
+            ));
+        }
+
+        $factory = new $factoryName;
+        return $factory;
+    }
 }
```

Reply:
hm, as I've looked into it yesterday, it looked like there were missing some aliases 🤔
I'll check that again when I find spare time. 
Thanks for re-checking.

Reply:
@boesing may I help further on this?